scripts/install/install_unix.sh

```diff
@@ -13,165 +13,638 @@
-            ;;
-        *)
-            echo "unknown"
-            ;;
-    esac
-}
-
-install_package() {
-    case "$package_type" in
-        deb)
-            dpkg -i "$out_file_path"
-            ;;
-        rpm)
-            # -U installs OR upgrades
-            # just -i will fail if previous versions exists
-            rpm -U "$out_file_path"
-            ;;
-        *)
-            echo "Installation failed: Could not determine the package manager to use for installation (package_type: ${package_type})"
-            exit 1
-            ;;
-    esac
-
-    if command -v systemctl > /dev/null 2>&1; then
-        systemctl enable --now observiq-otel-collector
+# This will install the package by downloading the archived collector,
+# extracting the binaries, and then removing the archive.
+install_package()
+{
+  banner "Installing observIQ OpenTelemetry Collector"
+  increase_indent
+
+  proxy_args=""
+  if [ -n "$proxy" ]; then
+    proxy_args="-x $proxy"
+    if [ -n "$proxy_user" ]; then
+      proxy_args="$proxy_args -U $proxy_user:$proxy_password"
     fi
+  fi
+
+  if [ -n "$proxy" ]; then
+    info "Downloading package using proxy..."
+  fi 
+
+  info "Downloading package..."
+  eval curl -L "$proxy_args" "$collector_download_url" -o "$out_file_path" --progress-bar --fail || error_exit "$LINENO" "Failed to download package"
+  succeeded
+
+  info "Extracting package..."
+  extract_package || error_exit "$LINENO" "Failed to extract package"
+  succeeded
+
+  info "Enabling service..."
+  systemctl enable --now observiq-otel-collector > /dev/null 2>&1 || error_exit "$LINENO" "Failed to enable service"
+  succeeded
+
+  success "observIQ OpenTelemetry Collector installation complete!"
+  decrease_indent
 }
 
-uninstall_package() {
-    if command -v dpkg > /dev/null 2>&1; then
-        if dpkg -l "observiq-otel-collector" > /dev/null 2>&1; then
-            desired_state="$(dpkg -l observiq-otel-collector | tail -n1 | cut -b 1)"
-            if [ "$desired_state" = "i" ]; then
-                echo "Existing DEB installation detected, uninstalling..."
-                dpkg -r "observiq-otel-collector"
-            else
-                # The package is already uninstalled;
-                # This case can be hit when the package was uninstalled, but config files are left behind,
-                # which dpkg still keeps track of.
-                echo "dpkg was detected, but no observiq-otel-collector package is currently installed, skipping..."
-            fi
-        else
-            echo "dpkg was detected, but no observiq-otel-collector package is currently installed, skipping..."
-        fi
-    fi
+extract_package()
+{
+  case "$package_type" in
+    deb)
+      dpkg -i "$out_file_path" > /dev/null
+      ;;
+    rpm)
+      rpm -U "$out_file_path" > /dev/null
+      ;;
+    *)
+      error "Unrecognized package type"
+      return 1
+      ;;
+  esac
+  return 0
+}
```

Comment:
This is a misnomer. This installs the package, not extract the package.

Reply:
yeah I know. There was already a build in function from the previous script called `install_package`. Got any other name suggestions? `unpack_package`?

Reply:
Ugh, I see what you mean. I think that other "install_package" function should be something like just "install". I mean, it's definitely doing more than just installing the package. Do we need to keep that other function name the same?